lib/srv/authhandlers.go

```diff
@@ -273,9 +275,28 @@ func (h *AuthHandlers) UserKeyAuth(conn ssh.ConnMetadata, key ssh.PublicKey) (*s
 	}
 	teleportUser := cert.KeyId
 
+	connectionDiagnosticID := cert.Extensions[teleport.CertExtensionConnectionDiagnosticID]
+
 	// only failed attempts are logged right now
 	recordFailedLogin := func(err error) {
 		failedLoginCount.Inc()
+
+		message := "Principal is not allowed by this certificate. Ensure your roles allows you use it."
```

Comment:
Also, can principal be included in the message?
```suggestion
		message := "Principal is not allowed by this certificate. Ensure your roles allows you to use it."
```